src/Elm/Parser/Expression.elm

```diff
@@ -25,60 +25,56 @@ expression =
     Pratt.expression
         -- TODO make sure that operators and expressions are indented properly
         { oneOf =
-            [ numberExpression
+            [ referenceExpression
                 |> Pratt.literal
-            , referenceExpression
-                |> Pratt.literal
-            , ifBlockExpression
-            , tupledExpression
-            , recordAccessFunctionExpression
-                |> Pratt.literal
-            , negationOperation
-            , letExpression
-            , lambdaExpression
             , literalExpression
                 |> Pratt.literal
-            , charLiteralExpression
+            , numberExpression
                 |> Pratt.literal
-            , recordExpression
+            , tupledExpression
             , glslExpression
                 |> Pratt.literal
             , listExpression
+            , recordExpression
             , caseExpression
+            , lambdaExpression
+            , letExpression
+            , ifBlockExpression
+            , recordAccessFunctionExpression
+                |> Pratt.literal
+            , negationOperation
+            , charLiteralExpression
+                |> Pratt.literal
             ]
         , andThenOneOf =
-            [ -- Elm Basics sets the precedence of <| and |> to 0, but Pratt.expression starts at 0 so it needs to be larger
-              infixRight 1 "<|"
-            , infixLeft 1 "|>"
-            , infixRight 2 "||"
-            , infixRight 3 "&&"
-
+            -- TODO Add tests for all operators
             -- TODO Report a syntax error when encountering multiple of the comparison operators
             -- `a < b < c` is not valid Elm syntax
-            -- TODO Add tests for all operators
-            , infixNonAssociative 4 "=="
-            , infixNonAssociative 4 "/="
-            , infixNonAssociative 4 "<="
-            , infixNonAssociative 4 ">="
+            [ recordAccess
+            , infixLeft 1 "|>"
```

Comment:
A lot of these operators start with a shared symbol, notably `|` and `<`. I'm wondering whether we have a parser a bit like
```elm
Parser.succeed identity
  |. Parser.symbol "|"
  |= Parser.oneOf
    [ Parser.succeed ( Left, 1 )
        |. Parser.symbol ">"
    , Parser.succeed ( Right, 2 )
        |. Parser.symbol "|"
    , ...
    ]
```

From what I can tell this is a bit involved because the `andThenOneOf` is supposed to have static precedence in order to filter them out based on the current precedence (which might actually be quite slow as well?)

(I'm just brainstorming here)

Reply:
Worth exploring, if the precedence is the same, then it could work. If it's different, I'm not sure...